types/overloads.d.ts

```diff
@@ -0,0 +1,661 @@
+
+  send(requestType: "StartStopReplayBuffer"): Promise<void>;
+
+  send(requestType: "StartReplayBuffer"): Promise<void>;
+
+  send(requestType: "StopReplayBuffer"): Promise<void>;
+
+  send(requestType: "SaveReplayBuffer"): Promise<void>;
+
+  send(
+    requestType: "SetCurrentSceneCollection",
+    args: { "sc-name": string }
+  ): Promise<void>;
+
+  send(
+    requestType: "GetCurrentSceneCollection"
+  ): Promise<{ "sc-name": string }>;
+
+  send(
+    requestType: "ListSceneCollections"
+  ): Promise<{ "scene-collections": string[] }>;
+
+  send(
+    requestType: "GetSceneItemProperties",
+    args: { scene?: string; item: { id: string; name: string } }
+  ): Promise<{
+    scene: string;
+    item: {
+      id: string;
+      rotation: number;
+      name: string;
+      visible: boolean;
+      locked: boolean;
+      bounds: { y: number; type: string; alignment: number; x: number };
+      scale: { x: number; y: number };
+      position: { x: number; y: number; alignment: number };
+      crop: { right: number; bottom: number; left: number; top: number };
+    };
+  }>;
+
+  send(
+    requestType: "SetSceneItemProperties",
+    args: {
+      scene?: string;
+      item: {
+        id: number;
+        rotation: number;
+        name: string;
+        visible: boolean;
+        locked: boolean;
+        bounds: { y: number; type: string; alignment: number; x: number };
+        scale: { x: number; y: number };
+        position: { x: number; y: number; alignment: number };
+        crop: { bottom: number; left: number; right: number; top: number };
+      };
+    }
+  ): Promise<void>;
+
+  send(
+    requestType: "ResetSceneItem",
+    args: { "scene-name"?: string; item: string }
+  ): Promise<void>;
+
+  send(
+    requestType: "SetSceneItemRender",
+    args: { source: string; render: boolean; "scene-name"?: string }
+  ): Promise<void>;
+
+  send(
+    requestType: "SetSceneItemPosition",
+    args: { "scene-name"?: string; item: string; x: number; y: number }
+  ): Promise<void>;
+
+  send(
+    requestType: "SetSceneItemTransform",
+    args: {
+      "scene-name"?: string;
+      item: string;
+      "x-scale": number;
+      "y-scale": number;
+      rotation: number;
+    }
+  ): Promise<void>;
+
+  send(
+    requestType: "SetSceneItemCrop",
+    args: {
+      "scene-name"?: string;
+      item: string;
+      top: number;
+      bottom: number;
+      left: number;
+      right: number;
+    }
+  ): Promise<void>;
+
+  send(
+    requestType: "SetCurrentScene",
+    args: { "scene-name": string }
+  ): Promise<void>;
+
+  send(
+    requestType: "GetCurrentScene"
+  ): Promise<{ name: string; sources: any[] }>;
+
+  send(
+    requestType: "GetSceneList"
+  ): Promise<{ "current-scene": string; scenes: any[] }>;
+
+  send(
+    requestType: "SetSceneItemOrder",
+    args: {
+      scene?: string;
+      items: any[];
+      "items[]": { id?: number; name?: string };
```

Comment:
So the correct format for this would be something like this?

```
items: { id?: number, name?: string }[];
```

Reply:
Ah yes! This must be a case that this parser is not yet correctly handling.

Reply:
Yeah, I see now where it works in `GetSourcesTypesList`. Probably just a matter of changing the formatting in the plugin docs.